numba/core/codegen.py

```diff
@@ -668,18 +673,58 @@ def _optimize_final_module(self):
         """
         Internal: optimize this library's final module.
         """
-        cheap_name = "Module passes (cheap optimization for refprune)"
-        with self._recorded_timings.record(cheap_name):
-            # A cheaper optimisation pass is run first to try and get as many
-            # refops into the same function as possible via inlining
-            self._codegen._mpm_cheap.run(self._final_module)
-        # Refop pruning is then run on the heavily inlined function
-        if not config.LLVM_REFPRUNE_PASS:
-            self._final_module = remove_redundant_nrt_refct(self._final_module)
-        full_name = "Module passes (full optimization)"
-        with self._recorded_timings.record(full_name):
-            # The full optimisation suite is then run on the refop pruned IR
-            self._codegen._mpm_full.run(self._final_module)
+        optimization_processors = {p for c in (global_processors(),
+                                               self._opt_info)
+                                   for p in c}
+        remarks_filter = None
+        remarks_file = None
+        remarks_data = []
+        try:
+            if optimization_processors:
+                remarks_filter = "|".join("(%s)" % f for p in
+                                          optimization_processors for f in
+                                          p.filters())
+                remarkdesc, remarks_file = mkstemp()
+                # Close file
+                with os.fdopen(remarkdesc, 'r'):
+                    pass
```

Comment:
couldn't we let llvmlite handle the file by using `run_with_remarks`?

Reply:
Not without having totally independent logic for the two paths and I'd rather not do that in case there are any changes to the pass manager logic.